pkg/security/admission/scc_exec.go

```diff
@@ -53,7 +54,7 @@ func (d *sccExecRestrictions) Admit(a admission.Attributes) (err error) {
 	// we're allowed to use the SA the pod is using.  Otherwise, user-A creates pod and user-B (who can't use the SA) can exec into it.
 	createAttributes := admission.NewAttributesRecord(pod, nil, kapi.Kind("Pod").WithVersion(""), a.GetNamespace(), a.GetName(), a.GetResource(), "", admission.Create, a.GetUserInfo())
 	if err := d.constraintAdmission.Admit(createAttributes); err != nil {
-		return admission.NewForbidden(a, fmt.Errorf("%s operation is not allowed because the pod's security context exceeds your permissions: %v", a.GetSubresource(), err))
```

Comment:
`NewForbidden` uses a metadata test to pull out name which is unnecessary.  A name can be retrieved from attributes.  If/when it fails, the errors get appended.  Fix `extractResourceName` to avoid object introspection where possible.  Where not possible, don't include the offending error, just return a fake name that indicates an error and is obviously not real.